### ISSUE TITLE
main() in test scope is executed twice when using test framework and a main module initializer in Test

### ISSUE THREAD

**Issue by cuzfrog** [scala-js/scala-js#3218]
when:
```scala
scalaJSUseMainModuleInitializer in Test := true
scalaJSModuleKind := ModuleKind.CommonJSModule
```
A main() in test will be executed twice by a single sbt `test:run`
```scala
object TmpApp {
  def main(args: Array[String]): Unit = {
    println("asdf")
  }
}
```
output:
```
[info] Running anywhere.TmpApp
asdf
asdf
```

scalajs version `0.6.21`

**Comment by sjrd** [comment]
I cannot reproduce the issue. With the files in [this gist](https://gist.github.com/sjrd/b3b562a2d7eed1e937ac0c7d371c3a7d), running `test:run` emits:
```
> test:run
[info] Compiling 1 Scala source to C:\Users\Sepi\Documents\Projets\test\target\scala-2.12\test-classes...
[info] Fast optimizing C:\Users\Sepi\Documents\Projets\test\target\scala-2.12\test-test-fastopt.js
[info] Running TmpApp
asdf
[success] Total time: 2 s, completed 09-dÚc.-2017 21:59:31
```
as expected.

**Comment by cuzfrog** [comment]
Test again in https://github.com/cuzfrog/scalajs-playground/tree/master/scalajs-app-test

If there's a test framework dependency, it occurs. But the `TmpApp` itself has nothing to do with any of the test frameworks. Comment out these dependencies, and it behaves normally.

**Comment by sjrd** [comment]
OK, I can reproduce it now. Note that `CommonJSModule` is not necessary for the reproduction.

Also, this does not happen in 1.0.0-M1+ anymore. It was fixed "accidentally" by this change:
https://github.com/scala-js/scala-js/commit/291d907264bcdf72ea53790e1eac710c7f095d76#diff-dd593403434ec9de3670cb5722f10171R752

We should be able to backport that specific line change to 0.6.x.

However, please note that, in any case, if you have `scalaJSUseMainModuleInitializer in Test := true`, then when running `test` (not `test:run`), your main method will *also* be run, and there's nothing you or we can do about it. Worse: it can be executed *multiple times* since we use multiple JS engines to run tests in parallel, and each one will run the main method.

So in general, the combination `scalaJSUseMainModuleInitializer in Test := true` + having a testing framework is something that does not make much sense, irrespective of the present issue.

**Comment by sjrd** [comment]
This was fixed in ce4004ee3caddee9a9622bf44f25a1f19e568720.
